### PR TITLE
Add links to morning challenge and D2 workshop

### PR DESCRIPTION
**Pull request by Akin909**
Add links to animated app drawer challenge and express routing and testing workshop.
Also update research afternoon readme with topic details. Relates #469 #467 
The workshops are available [here - app drawer](https://github.com/Akin909/morning-challenge-animated-app-drawer) and [here - testing with express](https://github.com/Akin909/express-and-testing-workshop)

### CODE REVIEW

**Comment by PiotrBerebecki** [comment]
@eliascodes, all the materials from week-8 will be moved to the `foundersandcoders` organisation

File: coursebook/week-8/README.md [hunk 1]
```diff
@@ -11,22 +11,23 @@ Schedule
...
 16:00-18:00 - Business Development / Community Engagement
 
 ### Day 2:
 
-10:00-11:00 - Morning Challenge
+10:00-11:00 - Morning Challenge - [Create an Animated App
+Drawer](https://github.com/Akin909/morning-challenge-animated-app-drawer)
 
-11:00-12:45 - Workshop 3
+11:00-12:45 - Workshop 3 - [Creating and Testing Express Routes](https://github.com/Akin909/express-and-testing-workshop)
```
Comment (des-des) [inline coursebook/week-8/README.md]: Cool, i have raised some issues against the repo

File: coursebook/week-8/research-afternoon.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+## Promises and Fetch
+- What is a Promise?
+- What is fetch and how do you use it?
+- What advantages do promises provide over callbacks
```
Comment (des-des) [inline coursebook/week-8/research-afternoon.md]: What are the disadvantages should also be in there as a question

File: coursebook/week-8/research-afternoon.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+## Express Middlewares
+- What are express middlewares?
+- What functionality do they provide?
+- What are some examples of useful express middleware and 
+how do you use them ([several userful examples](https://blog.jscrambler.com/setting-up-5-useful-middlewares-for-an-express-api/))
+
+## Session-management in Express
+- What are sessions?
+- How do you manage sessions in Express
```
Comment (des-des) [inline coursebook/week-8/research-afternoon.md]: I think:

`What are the different ways of managing sessions in express?` (students might get confused if we say there is only one answer)

File: coursebook/week-8/research-afternoon.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+- Create a minimal example of how to set up a session (FYI: pseudo code is
+  fine)
+
+## Handlebars tips and tricks
+- What are helper functions?
```
Comment (des-des) [inline coursebook/week-8/research-afternoon.md]: `What are helper functions in handlebars?`

File: coursebook/week-8/research-afternoon.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+- What are some examples of useful express middleware and 
+how do you use them ([several userful examples](https://blog.jscrambler.com/setting-up-5-useful-middlewares-for-an-express-api/))
+
+## Session-management in Express
+- What are sessions?
+- How do you manage sessions in Express
+- Create a minimal example of how to set up a session (FYI: pseudo code is
+  fine)
+
+## Handlebars tips and tricks
```
Comment (des-des) [inline coursebook/week-8/research-afternoon.md]: This list seems a bit random.

How about: what is server-side rendering, what is a templating language, what is handlebars .. ?

File: coursebook/week-8/README.md (hunk 1 above)
Comment (des-des) [inline coursebook/week-8/README.md]: Not a blocker on this pull, that convo can happen over there

Comment (Akin909) [inline coursebook/week-8/README.md]: I replied to those on the repo and fixed up the readme

File: coursebook/week-8/research-afternoon.md (hunk 2 above)
Comment (Akin909) [inline coursebook/week-8/research-afternoon.md]: I have a point about the holes in the native fetch which I think are important, I've also added something about the downsides of promises but personally haven't encountered any that are a specific problem with promises vs callbacks, did you have anything in mind?

File: coursebook/week-8/research-afternoon.md (hunk 5 above)
Comment (Akin909) [inline coursebook/week-8/research-afternoon.md]: Speaking for myself and anecdotally from most of my cohort we generally didn't get much out of research but the few things I did find particularly useful was seeing a basic implementation of a concept another group was trying to explain which is the idea behind this. I've swapped out the final section for server side rendering but I feel like a small example even copy pasted just giving them a quick glance at how sessions can be setup would be really helpful

File: coursebook/week-8/research-afternoon.md (hunk 3 above)
Comment (Akin909) [inline coursebook/week-8/research-afternoon.md]: Fixed

File: coursebook/week-8/research-afternoon.md (hunk 2 above)
Comment (des-des) [inline coursebook/week-8/research-afternoon.md]: Error handling is very different, and doing it properly can be more complex using promises.

**Comment by des-des** [comment]
ops was in QA mode merged without thinking someone else might want to review. Hope thats okay

**Pull request merged**